src/client/pythonEnvironments/common/commonUtils.ts

```
@@ -41,21 +39,26 @@ export async function* findInterpretersInDir(
         throw err; // re-throw
     }
 
-    const generators = dirContents.map((fullPath) => {
-        async function* generator() {
-            if (await isDirectory(fullPath)) {
-                if (recurseLevels && recurseLevels > 0) {
-                    yield* findInterpretersInDir(fullPath, recurseLevels - 1, filter);
-                }
-            } else if (checkBin(fullPath)) {
-                yield fullPath;
+    for (const entry of entries) {
+        const filename = path.join(root, entry.name);
+        if (itemFilter && !itemFilter(filename)) {
```

We should handle errors thrown by this so the next iteration wouldn't fail.

---

@ericsnowcurrently It probably makes sense to wrap each iteration of loop in a try..catch block.

---

It wasn't strictly necessary (relative to compatibility) but I've added the option `findInterpretersInDir()` to ignore errors.